compiler/injectdestructors.nim

```diff
@@ -412,7 +412,6 @@ template isExpression(n: PNode): bool =
   (not isEmptyType(n.typ)) or (n.kind in nkLiterals + {nkNilLit, nkRange})
 
 proc recurse(n: PNode, c: var Con, processProc: proc): PNode =
-  if n.sons.len == 0: return n
```

Comment:
@cooldome I'm redoing the refactoring. Can you explain why you had to have this line removed?